### PR TITLE
FWCore packages: cleanup use of GCC11Compatibility.h

### PR DESCRIPTION
**Pull request by gartung**
#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
part of https://github.com/cms-sw/cmssw/issues/28747

**Comment by gartung** [comment]
please test with #29122

**Comment by silviodonato** [comment]
please test

**Comment by silviodonato** [comment]
abort

**Comment by silviodonato** [comment]
please test with #29121

**Comment by gartung** [comment]
please test

**Comment by silviodonato** [comment]
> please test

@gartung perhaps you forget "with #29121"

**Comment by gartung** [comment]
@silviodonato Indeed I forgot to test with the other PR.

**Comment by gartung** [comment]
please test with 29121

**Comment by gartung** [comment]
abort

**Comment by gartung** [comment]
@silviodonato I need to do to rework this with a warning when the GCC11Compatibility.h is used.

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test with #29121

**Comment by makortel** [comment]
@cmsbuild, please test

https://github.com/cms-sw/cmssw/pull/29121 is now in IB

File: FWCore/Utilities/interface/GCC11Compatibility.h [hunk 1]
```diff
@@ -1,42 +1,4 @@
...
-#endif  // gcc 4.7
-#else
-#define constexpr
-#define noexcept
-#define nullptr 0
-#define GCC11_FINAL
-#define GCC11_OVERRIDE
-#endif  // NOCXX11
-
+#warning "GCC11Compatibility.h should be replaced with Visibility.h or Likely.h"
```
Comment (makortel) [inline FWCore/Utilities/interface/GCC11Compatibility.h]: AFAICT in the HEAD of master only files in this PR include `GCC11Compatibility.h`. Let's anyway put this PR in first, and then remove the header in a subsequent PR.

**Comment by makortel** [comment]
+1

Reco differences are from logErrorHarvester

**Comment by silviodonato** [comment]
+1

**Pull request merged**